### PR TITLE
Add example for let_* functions

### PR DESCRIPTION
**Pull request by lucteo**
Add both C++ code and bikeshed wording.

More examples with the server theme will follow.

### CODE REVIEW

**Comment by ericniebler** [comment]
These look good to me. Thanks, @lucteo. @kirkshoop, any thoughts?

File: examples/server_theme/let_value.cpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        | ex::let_error(error_to_response)
+        // If the flow is cancelled, send back a proper response
+        | ex::let_done(done_to_response)
+        // write the result back to the client
+        | ex::let_value(send_response)
+        // done
+        ;
+
+    // execute the whole flow asynchronously
+    ex::start_detached(std::move(snd));
```
Comment (kirkshoop) [inline examples/server_theme/let_value.cpp:147]: How do these examples exit safely?

The following request_stop call will cancel the work- but I do not see anything that waits for the work. 

This scenario is something that we would like to support with an async_scope in a forthcoming paper. 

For now there should be some way to wait for the pool to empty.

**Review by kirkshoop — COMMENTED** [review]
I love these samples. Especially the http example.

File: examples/server_theme/let_value.cpp (hunk 1 above)
Comment (kirkshoop) [reply examples/server_theme/let_value.cpp:147]: Never mind. I have trouble keeping track of all the static_thread_pool impls. I see that this one blocks in the destructor, which I deeply dislike.

**Review by kirkshoop — APPROVED** [review]
These will need to be used to show the benefits of async_scope.

**Pull request merged**